siamese_network/main.py

```diff
@@ -249,6 +249,8 @@ def main():
                         help='Learning rate step gamma (default: 0.7)')
     parser.add_argument('--no-cuda', action='store_true', default=False,
                         help='disables CUDA training')
+    parser.add_argument('--no-xpu', action='store_true', default=False,
```

Comment:
`default=False` does not make sense for `store_true`. I think it's better to drop it in the line you add and in the similar lines above and below.